core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -497,6 +497,10 @@ class GroupCoordinator(val brokerId: Int,
     }
   }
 
+  def handleDeletedPartitions(tpList: Seq[TopicPartition]) {
+    for (tp <- tpList) groupManager.cleanupGroupMetadata(Some(tp))
```

Comment:
Would it make sense to let `cleanupGroupMetadata` accept the full list? That way we can collect all of the tombstones that need to be written and do a single append to the log (for each group).